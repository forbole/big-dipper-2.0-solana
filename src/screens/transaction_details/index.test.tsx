/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { createMockClient } from 'mock-apollo-client';
import { ApolloProvider } from '@apollo/client';
import {
  MockTheme, wait,
} from '@tests/utils';
import {
TransactionDetailsDocument,
} from '@graphql/types';
import TransactionDetails from '.';

// ==================================
// mocks
// ==================================
jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {
      tx: 3,
    },
  }),
}));

const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('@components', () => ({
  Layout: (props) => <div id="Layout" {...props} />,
  LoadAndExist: (props) => <div id="LoadAndExist" {...props} />,
}));

jest.mock('./components', () => ({
  Overview: (props) => <div id="Overview" {...props} />,
  Messages: (props) => <div id="Messages" {...props} />,
}));

const mockTransactionDetailsDocument = jest.fn().mockResolvedValue({
  data: {
    "transaction": [
      {
        "slot": 112756074,
        "error": false,
        "signature": "5oNqh9epbE8AmCqVUr2hb238dyoqBKninb8cyMwwqgTQ5dckAkSqaA3JVhHm8ZK7AjcFp2X1PwvSyVbYGdqBLYGm",
        "fee": 5000,
        "block": {
          "timestamp": "2021-12-22T14:33:14"
        },
        "messages": [
          {
            "type": "vote"
          }
        ]
      }
    ]
  },
});

// ==================================
// unit tests
// ==================================
describe('screen: Blocks/List', () => {
  it('matches snapshot', async () => {
    const mockClient = createMockClient();
    mockClient.setRequestHandler(
      TransactionDetailsDocument,
      mockTransactionDetailsDocument,
    );

    let component;
    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={mockClient}>
          <MockTheme>
            <TransactionDetails />
          </MockTheme>
        </ApolloProvider>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
