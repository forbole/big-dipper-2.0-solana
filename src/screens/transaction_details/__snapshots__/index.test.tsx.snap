// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Blocks/List matches snapshot 1`] = `
<div
  id="Layout"
  navTitle="transactionDetails"
>
  <div
    exists={true}
    id="LoadAndExist"
    loading={false}
  >
    <span
      className="makeStyles-root"
    >
      <div
        data={
          Object {
            "fee": Object {
              "baseDenom": "udaric",
              "displayDenom": "daric",
              "exponent": 6,
              "value": "0.015",
            },
            "signature": "47vcSu9mH3mUPo5FRbv5dKkeDGCEV26H2V3GWSk25F8SrWwEFzcBvC9u1DpEJwx1k5apYF6uHL1Nv8PFZBZGzwzn",
            "slot": 126542151,
            "success": false,
            "timestamp": "2022-03-25T02:30:13",
          }
        }
        id="Overview"
      />
      <div
        id="Logs"
        logs={
          Array [
            "Program 11111111111111111111111111111111 invoke [1]",
            "Program 11111111111111111111111111111111 success",
          ]
        }
      />
    </span>
  </div>
</div>
`;
